src/client.ts

```diff
@@ -185,14 +185,22 @@ export class Client {
     this.#protocol.socket.onclose = () => pList.at(-1)?.resolve();
 
     // Close browser process (also closes the ws endpoint, which in turn closes all sockets)
-    this.#browser_process.stderr!.close();
-    this.#browser_process.stdout!.close();
-    this.#browser_process.close();
+    if (this.#browser_process) {
+      this.#browser_process.stderr!.close();
+      this.#browser_process.stdout!.close();
+      this.#browser_process.close();
+    } else {
+      //When Working with Remote Brwosers, where we don't control the Browser Process explicitly
+      await this.#protocol.send("Browser.close");
+    }
 
     // Zombie processes is a thing with Windows, the firefox process on windows
     // will not actually be closed using the above.
     // Related Deno issue: https://github.com/denoland/deno/issues/7087
-    if (this.browser === "firefox" && Deno.build.os === "windows") {
+    if (
+      this.#browser_process && this.browser === "firefox" &&
+      Deno.build.os === "windows"
+    ) {
```

Comment:
actually, do you mind commenting out this conditional and its content? realised i must have missed it